### PR TITLE
add yellowfinLogparser and yellowfin log to go code

### PR DESCRIPTION
**Pull request by pnyengan**
Please review my code

### CODE REVIEW

**Comment by bmharper** [comment]
All Go code must be formatted with gofmt. Just install and use SublimeText with GoSublime. It really is easy. There's no excuse not to do this.

Also, the comment showing a sample Yellowfin log message is misleading: `//YF:2015-11-24 16:40:47: INFO`. In that comment, why do you include the "YF:" prefix? That prefix doesn't exist in a real Yellowfin log message. It is inconsistent with the rest of the code in that function.

Another thing - in those comments, you must include an example of an INFO message as well as an ERROR message, so that it is clear how Yellowfin's log messages use inconsistent spacing for ERROR vs INFO. Then also, you must strip the leading space off the INFO messages. I assume that other error messages with 4 letter lengths will also need to be stripped of their leading spaces.

**Comment by pnyengan** [comment]
code review changes done. Please review my code.

**Comment by bmharper** [comment]
This looks good. One _final_ thing that I've just noticed now, is that the Yellowfin log dates have no timezone. Can you verify that the timestamps on the yellowfin logs emitted by the logscraper are correct?

By setting

``` go
s.SendToLoggly = false
```

and making sure that you have a mix of freshly generated router logs and yellowfin logs, you should be able to do a few spot checks on the JSON emitted, and verify that the dates are close together. South Africa time is 2 hours away from GMT, so if there is something wrong, you should expect to see a two hour difference between the router JSON messages and the yellowfin JSON messages.

**Comment by bmharper** [comment]
One more thing - the severity is not being trimmed. You must trim spaces off severity, otherwise and INFO severity will be represented at " INFO" (ie a space at the start).

File: parsers.go [hunk 1]
```diff
@@ -85,6 +88,22 @@ func RouterLogParser(msg []byte) *LogMsg {
...
 }
 
+func YellowfinLogParser(msg []byte) *LogMsg {
+	matches := yellowfinLogRegex.FindSubmatchIndex(msg)
+	if len(matches) != (4+1)*2 {
+		return nil
+	}
+	var err error
+	m := &LogMsg{}
+	m.Time, err = time.ParseInLocation(timeYellowfin, string(getCapture(msg, matches, 1)), time.Now().Location())
```
Comment (bmharper) [inline parsers.go]: Instead of `time.Now().Location()`, use `time.Local` to retrieve the local timezone.

**Comment by pnyengan** [comment]
@bmharper  Code review changes made.

**Pull request merged**